app/src/main/java/com/keylesspalace/tusky/ViewMediaActivity.kt

```diff
@@ -137,7 +143,19 @@ class ViewMediaActivity : BaseActivity(), ViewImageFragment.PhotoActionsListener
             true
         }
 
-        window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LOW_PROFILE
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+
+        // make sure the toolbar does not move when the system bars are hidden
+        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { _, insets ->
+            val top = insets.getInsets(systemBars()).top
+            val toolbarParams = binding.toolbar.layoutParams as ViewGroup.MarginLayoutParams
+            toolbarParams.topMargin = top
+            WindowInsetsCompat.CONSUMED
+        }
+        // hide the system bars
+        WindowInsetsControllerCompat(window, window.decorView).run {
+            hide(systemBars())
+        }
```

Comment:
this isn't as nice as the `SYSTEM_UI_FLAG_LOW_PROFILE` unfortunately but looks there is no 100% replacement for it :(